python/lsst/analysis/ap/nb_utils.py

```diff
@@ -70,3 +77,222 @@ def make_simbad_link(ra, dec, radius_arcsec=3.0):
+        any power of 10, or omitted entirely.
+    chunk_size2: `int`, optional
+        Chunk size to use for output directories of cutouts unique to
+        dataset 2. If using an existing set of cutouts made with
+        plotImageSubtractionCutouts, be sure to set this value to
+        match what was used when making the set. Then the code will
+        detect the existing files and not attempt to remake them.
+        If not using a preexisting set, this value can be set to
+        any power of 10, or omitted entirely.
+    njobs: `int`, optional
+        Number of parallel processes for
+        plotImageSubtractionCutouts (default 1)
+
+    Returns
+    -------
+
+    unique1: `pandas.DataFrame`
+        Data frame of sources only found in the first dataset.
+    unique2: `pandas.DataFrame`
+        Data frame of sources only found in the second dataset.
+    matched: `pandas.DataFrame`
+        Data frame of matched sources; actually only the sources from
+        the first dataset but with a new column pointing to the
+        diaSourceId of the match in the decond dataset.
+    """
+
+    if makeCutouts and cutoutPath1 is None:
+        errstr = ('You must supply a value for cutoutPath1 '
+                  'and cutoutPath2 if makeCutouts is True.')
+        raise ValueError(errstr)
+    if makeCutouts and cutoutPath2 is None:
+        errstr = ('You must supply a value for cutoutPath1 '
+                  'and cutoutPath2 if makeCutouts is True.')
+        raise ValueError(errstr)
+
+    if badFlagList is not None:
+        query1.set_excluded_diaSource_flags(badFlagList)
+        query2.set_excluded_diaSource_flags(badFlagList)
+    goodSrc1 = query1.load_sources(exclude_flagged=True)
+    goodSrc2 = query2.load_sources(exclude_flagged=True)
+
+    if 'reliability' not in goodSrc1.columns:
+        goodSrc1['reliability'] = None
+    if 'reliability' not in goodSrc2.columns:
+        goodSrc2['reliability'] = None
+
+    """
+    We assume that the runs to be compared here will always be
+    over the same dataset with the same visit and detector numbers.
+    Then we can enforce the matching on that level, too.
+    We could relax that assumption later, though it would require
+    removing the visit, detector loop.
+    """
+
+    visit_det = set(zip(goodSrc1['visit'], goodSrc1['detector']))
+    src1 = []
+    src2 = []
+    matches = []
+    indices = {}
+    seps = {}
+
+    for visit, detector in visit_det:
+        mask1 = (goodSrc1['visit'] == visit) &\
+            (goodSrc1['detector'] == detector)
+        mask2 = (goodSrc2['visit'] == visit) &\
+            (goodSrc2['detector'] == detector)
+        gs1 = goodSrc1[mask1].copy()
+        gs2 = goodSrc2[mask2].copy()
+
+        coords1 = SkyCoord(ra=gs1['ra'].values*u.degree,
+                           dec=gs1['decl'].values*u.degree)
+        coords2 = SkyCoord(ra=gs2['ra'].values*u.degree,
+                           dec=gs2['decl'].values*u.degree)
```

Comment:
Shouldn't this be `dec` now, and not `decl`?

Reply:
It sure as heck should be, good catch! I see now that in my last testing I had forgotten to switch to my most recent diffim sprint dataset runs, which were after we made that change. My earlier tests were on older runs.